src/Adapter/OrderReturn/Repository/OrderReturnRepository.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\OrderReturn\Repository;
+
+use OrderReturn;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnException;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\ValueObject\OrderReturnId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\OrderReturn\Repository\OrderReturnRepositoryInterface;
+use PrestaShop\PrestaShop\Core\Repository\AbstractObjectModelRepository;
+
+class OrderReturnRepository extends AbstractObjectModelRepository implements OrderReturnRepositoryInterface
+{
+    /**
+     * Gets legacy OrderReturn
+     *
+     * @param OrderReturnId $orderReturnId
+     *
+     * @return OrderReturn
+     *
+     * @throws OrderReturnException
+     * @throws CoreException
+     */
+    public function get(OrderReturnId $orderReturnId): OrderReturn
```

Comment:
Should return an object an `OrderReturnInterface` that implements all the getters , so we don't have legacy code in the new code.

Reply:
Don't do it just yet. At least not in this PR. There are plenty of this kind repositories already and none of them yet have an interface for object model (also the abstract repository methods won't be compatable with the dedicated interfaces). These repositories were created with the Adapter namespace in mind and were expected to deal with the ObjectModel directly. (then the ADR of removing the Adapter was introduced).
If we want to start introducing interfaces for ObjectModels, then we need to think about it more. (think of what makes sense to have in such interface and what not, because current legacy ObjectModels has lots of crazy methods and I don't think we want to create contracts for all of them). If not thought through, this might lead into yet another mess.

Reply:
Does the same apply to Repository interfaces or should I keep them?